File: app/src/main/java/com/tronprotocol/app/selfmod/ReflectionResult.java
package com.tronprotocol.app.selfmod;

import java.util.ArrayList;
import java.util.List;

/**
 * Result of self-reflection on behavior
 */
public class ReflectionResult {
    private final List<String> insights;
    private final List<String> suggestions;
    
    public ReflectionResult() {
        this.insights = new ArrayList<>();
        this.suggestions = new ArrayList<>();
    }
    
    public void addInsight(String insight) {
        insights.add(insight);
    }
    
    public void addSuggestion(String suggestion) {
        suggestions.add(suggestion);
    }
    
    public List<String> getInsights() {
        return new ArrayList<>(insights);
    }
    
    public List<String> getSuggestions() {
        return new ArrayList<>(suggestions);
    }
    
    public boolean hasInsights() {
        return !insights.isEmpty();
    }
    
    public boolean hasSuggestions() {
        return !suggestions.isEmpty();
    }
    
    @Override
    public String toString() {
        return "ReflectionResult{" +
                "insights=" + insights.size() +
                ", suggestions=" + suggestions.size() +
                '}';
    }
}

File: app/src/main/java/com/tronprotocol/app/selfmod/ReflectionResult.kt
package com.tronprotocol.app.selfmod

/**
 * Result of self-reflection on behavior
 */
class ReflectionResult {
    private val insights = mutableListOf<String>()
    private val suggestions = mutableListOf<String>()

    fun addInsight(insight: String) {
        insights.add(insight)
    }

    fun addSuggestion(suggestion: String) {
        suggestions.add(suggestion)
    }

    fun getInsights(): List<String> = ArrayList(insights)

    fun getSuggestions(): List<String> = ArrayList(suggestions)

    fun hasInsights(): Boolean = insights.isNotEmpty()

    fun hasSuggestions(): Boolean = suggestions.isNotEmpty()

    override fun toString(): String {
        return "ReflectionResult{" +
                "insights=${insights.size}" +
                ", suggestions=${suggestions.size}" +
                "}"
    }
}
